include/flyft/three_dimension_index.h

```diff
@@ -0,0 +1,21 @@
+#ifndef FLYFT_THREE_DIMENSION_INDEX_H_
+#define FLYFT_THREE_DIMENSION_INDEX_H_
+
+#include<cmath>
+
+namespace flyft
+{
+class ThreeDimensionIndex
+    {
+    public:
+        ThreeDimensionIndex(int ni, int nj, int nk);
+        int size() const;
+        int operator()(int i, int j, int k) const;
+    protected:
```

Comment:
These can be private, we aren't likely to subclass this
```suggestion
    private:
```

Reply:
This comment still applies